### PR TITLE
Fix for amend-return journey

### PR DESCRIPTION
**Pull request by lmcewanhmrc**

### CODE REVIEW

File: app/uk/gov/hmrc/plasticpackagingtaxreturns/controllers/ReturnsController.scala [hunk 1]
```diff
@@ -78,7 +78,7 @@ class ReturnsController @Inject()(
   def amend(pptReference: String): Action[AnyContent] =
     authenticator.authorisedAction(parse.default, pptReference) { implicit request =>
       getUserAnswer(request)(userAnswer =>  {
-        isDirectDebitInProgress(pptReference, userAnswer.get(PeriodKeyGettable).getOrElse(throw new Exception("Cannot amend return without period Key")))
+        isDirectDebitInProgress(pptReference, userAnswer.get[String](JsPath \ "amendSelectedPeriodKey").getOrElse(throw new Exception("Cannot amend return without period Key")))
```
Comment (liampan) [inline app/uk/gov/hmrc/plasticpackagingtaxreturns/controllers/ReturnsController.scala:81]: is it worth Renaming PeriodKeyGettable to ReturnPeriodKeyGettable and creating a AmendPeriodKeyGettable

File: app/uk/gov/hmrc/plasticpackagingtaxreturns/controllers/CalculationsController.scala [hunk 2]
```diff
@@ -44,16 +44,13 @@ class CalculationsController @Inject()(
...
         sessionRepository.get(request.cacheKey).map { optUa => {
-          for {
-            userAnswers <- optUa
-            original <- OriginalReturnForAmendValues(userAnswers)
-            amend <- AmendReturnValues(userAnswers)
-            originalCalc = calculationsService.calculate(original)
-            amendCalc = calculationsService.calculate(amend)
-          } yield
-            Json.toJson(AmendsCalculations(original = originalCalc, amend = amendCalc))
-        }.fold(UnprocessableEntity("No user answers found"))(Ok(_))
```
Comment (liampan) [inline app/uk/gov/hmrc/plasticpackagingtaxreturns/controllers/CalculationsController.scala:55]: probs doesnt matter, but we wont get UnprocessableEntity now we will get 500

**Review by liampan — APPROVED** [review]

File: app/uk/gov/hmrc/plasticpackagingtaxreturns/controllers/CalculationsController.scala (hunk 2 above)
Comment (liampan) [reply app/uk/gov/hmrc/plasticpackagingtaxreturns/controllers/CalculationsController.scala:55]: not sure actually what this change does 🤔

File: app/uk/gov/hmrc/plasticpackagingtaxreturns/controllers/ReturnsController.scala (hunk 1 above)
Comment (lmcewanhmrc) [reply app/uk/gov/hmrc/plasticpackagingtaxreturns/controllers/ReturnsController.scala:81]: Can do. Some of those gettables etc look like overkill to me - would prefer a comprehensive list of all the strings used user answers... and to share that file between front and back 😆

File: app/uk/gov/hmrc/plasticpackagingtaxreturns/controllers/CalculationsController.scala (hunk 2 above)
Comment (lmcewanhmrc) [reply app/uk/gov/hmrc/plasticpackagingtaxreturns/controllers/CalculationsController.scala:55]: True. Was easier to find the source of the problem like this 🤷

**Pull request merged**